internal/validations/observability/outputs/splunk.go

```diff
@@ -0,0 +1,19 @@
+package outputs
+
+import (
+	"fmt"
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	"strings"
+)
+
+func ValidateSplunk(spec obs.OutputSpec) (results []string) {
+	if spec.Splunk.PayloadKey != "" && spec.Splunk.IndexedFields != nil {
+		payload := string(spec.Splunk.PayloadKey)
+		for _, v := range spec.Splunk.IndexedFields {
+			if !strings.HasPrefix(string(v), payload) {
+				results = append(results, fmt.Sprintf("Indexed field: %s not part of payload: %s", v, payload))
```

Comment:
are we trying to restrict they payload field from being part of indexed fields?  If so then the error should read that it can not or is not allowed

Reply:
Indexed fields can't be outside of a given payload

Reply:
I don't believe this to be a reasonable restriction even though it is the behavior of the encoder.  Use the audit event example from the feature request where they are indexing annotations but the "event"/payload field comes from a sibling node in the original message.  We should be able to copy the fields in question to the "root" of the message so the encoder picks them up while setting the payload to something else.

Reply:
Maybe you are right, this checks are overhead and not clear. I will remove this